### ISSUE TITLE
Vulnerability: Solana on-chain program developer information disclosure

### ISSUE THREAD

**Issue by ant4g0nist** [solana-labs/solana#23354]
#### Problem
This vulnerability allows anyone to disclose the username of the development machine being used for development from the deployed program on the chain. This can be treated as a privacy concern for developers as they are not aware that this is happening. This information can be used to find the user on social media platforms and reveal their true identities.

The specific flaw exists when compiling the contract using `cargo build-bpf` command that uses `clang` in turn that embeds `hostname:pathname` in the `DW_AT_comp_dir` attributes in the final compiled ELF program. An attacker can leverage this vulnerability to disclose users behind the deployed contract which is not expected behavior on a blockchain.

#### Proposed Solution
It is recommended to remove the debug info including `DW_AT_comp_dir` attribute from the compiled programs on the chain before deploying. 

#### Proof-of-concept:
Attached to this email you can find a proof-of-concept python script that scrapes the Solana chain using Solscan and extracts user details from the executable program accounts.

Usage: ./dumper.py

```
root@54f6d22cc6c5:/home# ./dumper.py 
**********
Account: 4tacGjTExC28MiAMnYD4X3pe3LzR9xovLevTk2kjQF1w
Found users: runner,mgild
**********
Account: 3vYwnGLQFFLnpWwomB88YStFz3HtytND8NDyEhfXy7fA
Found users: runner,andrej
**********
Account: 4hByTbG1p8ZAydsGQjs2TW42wZzGep6UHhFZKZzNAyaA
Found users: runner,siong
**********
Account: 7nYabs9dUhvxYwdTnrWVBL9MYviKSfrEbdWCUbcnwkpF
Found users: runner,mgild
**********
Account: FYd6akA5NgW5PM6ZnAKv8rFQ38voaNgXj1CYP9GMUSDy
Found users: hareeshnagaraj
**********
Account: C11DCb9decDPgRUTF5GRd3H3KciQ7pCxWuMfhzFKEYrJ
Found users: runner,hareeshnagaraj
**********
Account: 52xHuqibnMfqrPeN2zpg1ygKkLpJ3HHxU1QzX97VB4AN
Found users: runner,davidcao
...
```

[dumper.py](https://gist.github.com/ant4g0nist/40a7e301922ce1d6cbd2150aea19acd5)

Since the issue is not critical, filing the issue here as per the response from security@solana.com

**Comment by t-nelson** [comment]
Please replace the screenshot with
```
inline text
```

**Comment by jackcmay** [comment]
@dmakarov Can you close this issue with a pointer of what PRs addressed it?

**Comment by dmakarov** [comment]
This issue should be resolved by #25881 and #25845.
